akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -75,6 +78,22 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
     activeStreamCount() < maxConcurrentStreams
   }
 
+  private var completing = false
+
+  /**
+   * Marks this object to be shutting down. If the buffer is empty, it also invokes [[onComplete()]].
+   */
+  def tryComplete(): Unit = {
+    completing = true
+    if (streamStates.isEmpty) onComplete()
+  }
+
+  /**
+   * Subclasses must override this with the code to run when all streams in StreamHandling
+   * buffers are closed
+   */
+  def onComplete()
```

Comment:
this looks like an event handler that is called when the stage completes, but actually it is the function that does the completing. Perhaps call it `complete()`?